### DISCUSSION TITLE
No iterator for contract_details

### DISCUSSION THREAD

**Discussion by BigKahunaCanuck** [wboayue/rust-ibapi#165] [General]
I would like to get the primary exchange for a stock symbol.  This would allow me to cut down on the amount of data retrieved (only one exchange versus all available exchanges) for market data requests.

**Comment by wboayue** [comment]
You can use `contract_details` to determine the primary exchange. From the `ContractDetails` struct your can find the primary exchange as follows.
```rust
contract_details.contract.primary_exchange
``` 
After you've done this once you can set the primary exchange when requesting contracts details to only receive data for that exchange.
```rust
let mut contract = Contract::stock("AAPL");   // this defaults exchange to SMART
contract.exchange = "NYSE";
client.contract_details(&contract);
```
